src/frontend/screens/AuthScreen.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from "react";
+import { defineMessages, FormattedMessage } from "react-intl";
+import { View, Image, Text, StyleSheet, BackHandler } from "react-native";
+
+import { DARK_BLUE, RED, WHITE } from "../lib/styles";
+import { SecurityContext } from "../context/SecurityContext";
+import { PasscodeInput } from "../sharedComponents/PasscodeInput";
+import { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { AppStackList } from "../Navigation/AppStack";
+import { useFocusEffect } from "@react-navigation/native";
+
+const m = defineMessages({
+  enterPass: {
+    id: "screens.EnterPassword.enterPass",
+    defaultMessage: "Enter your passcode",
+  },
+  wrongPass: {
+    id: "screens.EnterPassword.wrongPass",
+    defaultMessage: "Incorrect passcode, please try again ",
+  },
+});
+
+export const AuthScreen = ({
+  navigation,
+}: NativeStackScreenProps<AppStackList, "AuthScreen">) => {
+  const [error, setError] = React.useState(false);
+  const { authenticate, authState } = React.useContext(SecurityContext);
+  const [inputtedPass, setInputtedPass] = React.useState("");
+
+  const disableBack = React.useCallback((e: any) => {
+    e.preventDefault();
+  }, []);
+
+  const removeListener = React.useCallback(() => {
+    navigation.removeListener("beforeRemove", disableBack);
+  }, [navigation]);
+
+  React.useEffect(() => {
+    if (authState === "unauthenticated") {
+      navigation.addListener("beforeRemove", disableBack);
+    }
+  }, [authState, navigation, disableBack]);
+
+  function removeListenerAndGoBack() {
+    removeListener();
+    navigation.goBack();
+  }
```

Comment:
I am passing the `removeListenerAndGoBack()` directly to the `authenticate` function in the `securityContext` @gmaclennan